### PR TITLE
address points from issue #81 - 3dsmax2021 + vray5 changes

### PR DESCRIPTION
**Pull request by davidkydd**

### CODE REVIEW

File: templates/3dsmax/scripts/RepathRenderElements.ms [hunk 1]
```diff
@@ -21,28 +21,24 @@ if(rem != undefined) then
...
-if (indexVrayAdv == 1 or indexVrayNext == 1) then
+if (indexVRay5 == 1) then
 (
+    tokens = filterString irAutoSavePath "\\"
+    filename = tokens[tokens.count]
+    prefix = sysEnv.GetEnvironmentVariable "AZ_BATCH_TASK_WORKING_DIR"
+    newPath = prefix + "\\images\\" + filename
+
     irAutoSavePath = r.adv_irradmap_autoSaveFileName
     if(irAutoSavePath != undefined) then
```
Comment (pierre-augeard-nfinite) [inline templates/3dsmax/scripts/RepathRenderElements.ms]: Watch out for the case where `irAutoSavePath` is defined as a string but empty

so in the next line `tokens` will be undefined
```
tokens = filterString irAutoSavePath "\\"
```

A fix would be
```
if(irAutoSavePath != undefined and irAutoSavePath != "") then
```

File: templates/3dsmax/scripts/RepathRenderElements.ms [hunk 2]
```diff
@@ -21,28 +21,24 @@ if(rem != undefined) then
...
     (
-        tokens = filterString irAutoSavePath "\\"
-        filename = tokens[tokens.count]
-        prefix = sysEnv.GetEnvironmentVariable "AZ_BATCH_TASK_WORKING_DIR"
-        newPath = prefix + "\\images\\" + filename
         r.adv_irradmap_autoSaveFileName = newPath
     )
 
     irSavePath = r.adv_irradmap_saveFileName
     if(irSavePath != undefined) then
```
Comment (pierre-augeard-nfinite) [inline templates/3dsmax/scripts/RepathRenderElements.ms]: Watch out for the case where `irSavePath` is defined as a string but empty

so in the next line `tokens` will be undefined
```
tokens = filterString irSavePath "\\"
```

A fix would be
```
if(irSavePath != undefined and irSavePath != "") then
```

File: templates/3dsmax/scripts/RepathRenderElements.ms [hunk 3]
```diff
@@ -21,28 +21,24 @@ if(rem != undefined) then
...
 r = renderers.current     
 rendererName = r as string
-indexVrayAdv = findString rendererName "V_Ray_Adv_"
-indexVrayNext = findString rendererName "V_Ray_Next_"
+indexVRay5 = findString rendererName "V_Ray_5_"
 
-if (indexVrayAdv == 1 or indexVrayNext == 1) then
+if (indexVRay5 == 1) then
 (
+    tokens = filterString irAutoSavePath "\\"
```
Comment (pierre-augeard-nfinite) [inline templates/3dsmax/scripts/RepathRenderElements.ms]: ìrAutoSavePath` is not defined here

`newPath` is used at two different places after but there should be two different values (one for `irAutoSavePath`, one for `irSavePath`)

**Comment by pierre-augeard-nfinite** [comment]
Hello,
All the "dr" jobs use in for command line (calling 3dsmax.ps1) the parameter `-renderer [parameters('vrayRenderer')`
Value are still `VRayAdv` or `VRayNext` It's no compatible with the proposal code in 3dsmax.ps1 (well actually it was already non functional). Within 3dsmax.ps1 only the value "vray5" is supported
As consequence, the pool.template files for dr rendering should be modified.

File: templates/3dsmax/scripts/PrepareVrayServices.bat [hunk 4]
```diff
@@ -5,35 +5,10 @@ set VRAY_PORT=%3
...
-
-    IF "%MAX_VERSION%"=="2018" (
-	   start "vrayspawner2018" "C:\Autodesk\3dsMax2018\vrayspawner2018.exe" "-port=%VRAY_PORT%" > vrayexe.output.txt
-    ) 
-    IF "%MAX_VERSION%"=="2019" (
-	   start "vrayspawner2019" "C:\Autodesk\3dsMax2019\vrayspawner2019.exe" "-port=%VRAY_PORT%" > vrayexe.output.txt
-    )
-	IF "%MAX_VERSION%"=="2020" (
-	   start "vrayspawner2020" "C:\Autodesk\3dsMax2020\vrayspawner2020.exe" "-port=%VRAY_PORT%" > vrayexe.output.txt
+IF /I "%VRAY_RENDERER%"=="vray5" (
```
Comment (pierre-augeard-nfinite) [inline templates/3dsmax/scripts/PrepareVrayServices.bat]: `%VRAY_RENDERER%` comes from pool parameter `vrayRenderer` with value "VRay5", not "vray5"

File: templates/3dsmax/scripts/3dsmax.ps1 [hunk 5]
```diff
@@ -5,7 +5,7 @@ param (
     [string]$sceneFile,
     [int]$nodeCount = 1,
     [switch]$dr,
-    [string]$renderer = "vray",
+    [string]$renderer = "vray5",
```
Comment (pierre-augeard-nfinite) [inline templates/3dsmax/scripts/3dsmax.ps1]: from my understanding, the `$renderer` variable is always set in the job command line, and comes from the job "renderer" parameter
for instance
```
"commandLine": "powershell.exe 3dsmax.ps1 -maxVersion \\\"[parameters('maxVersion')]\\\" -start {0} -end {0} -outputName \\\"images\\[parameters('outputName')]\\\" -sceneFile \\\"[parameters('sceneFile')]\\\" -pathFile \\\"[parameters('pathFile')]\\\" -renderer [parameters('renderer')] -preRenderScript \\\"[parameters('preRenderScript')]\\\" -camera \\\"[parameters('camera')]\\\" -additionalArgs \\\" [parameters('additionalArgs')]\\\" -irradianceMap \\\"[parameters('irradianceMapFile')]\\\" -renderPresetFile \\\"[parameters('renderPresetFile')]\\\"",
```

In the job template, the allowed values are either "arnold" or "vray", never "vray5"

Comment (davidkydd) [reply templates/3dsmax/scripts/3dsmax.ps1]: yes this was a bit confusing and overloaded in prior templates - some templates, like those for VRay DR - have both a renderer parameter (set to vray) and a vrayRenderer param (set to vrayAdv, VrayRT). Sometimes the value of vrayRenderer was passed in as the value to the call to "3dsmax.ps1 -renderer $value",

File: templates/3dsmax/scripts/3dsmax.ps1 [hunk 6]
```diff
@@ -162,23 +98,40 @@ if ($renderer -eq "arnold")
...
-    $pre_render_script_content += "if (indexVrayNext == 1 and r.output_saveRawFile and r.output_rawFileName != """") then (fileName = ""$outputPath"" + (filenameFromPath r.output_rawFileName); r.output_rawFileName = fileName)`r`n"
-
-    $pre_render_script_content += "indexVrayNextGpu = findString rendererName ""V_Ray_GPU_Next_""`r`n"
-    $pre_render_script_content += "if (indexVrayNextGpu == 1 and r.V_Ray_settings.output_splitgbuffer and r.V_Ray_settings.output_splitfilename != """") then (fileName = ""$outputPath"" + (filenameFromPath r.V_Ray_settings.output_splitfilename); r.V_Ray_settings.output_splitfilename = fileName)`r`n"
-    $pre_render_script_content += "if (indexVrayNextGpu == 1 and r.V_Ray_settings.output_saveRawFile and r.V_Ray_settings.output_rawFileName != """") then (fileName = ""$outputPath"" + (filenameFromPath r.V_Ray_settings.output_rawFileName); r.V_Ray_settings.output_rawFileName = fileName)`r`n"
+    $vrayLogFile = "$env:AZ_BATCH_TASK_WORKING_DIR\VRayLog.log" -replace "\\", "\\"
+
+    $pre_render_script_content += "r.system_vrayLog_level = 4`r`n"
+    $pre_render_script_content += "r.system_vrayLog_file = ""$vrayLogFile""`r`n"
```
Comment (pierre-augeard-nfinite) [inline templates/3dsmax/scripts/3dsmax.ps1:108]: In previous version - without DR activated - vray log file was located at its default location `$env:LOCALAPPDATA\Temp\vraylog.txt`

1. This pattern is written in job templates
```
{
                            "filePattern": "vraylog.*",
                            "destination": {
                                "autoStorage": {
                                    "fileGroup": "[parameters('outputs')]",
                                    "path": "[parameters('jobName')]/logs/frame_{0}"
                                }
                            },
                            "uploadOptions": {
                                "uploadCondition": "taskCompletion"
                            }
                        }
```

2. Then the default file location is copied at the end of the 3dsmax.ps1 script
```
Copy-Item "$env:LOCALAPPDATA\Temp\vraylog.txt" . -ErrorAction SilentlyContinue
```

**Pull request merged**